contents/blog/persons-on-events.md

```diff
@@ -0,0 +1,36 @@
+---
+date: 2022-08-29
+title: 'How we’re improving performance by combining persons and events’'
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-array-blog.png
+featuredImageType: full
+---
+
+In [a previous product update](/blog/the-posthog-array-1-39-0) we announced a beta for a substantial change to the way we handle persons and events on PostHog. Today, after gathering your feedback and seeing the improvements to query performance, we’re rolling this change out to all PostHog Cloud users, and making it available for Self-Hosted users. 
+
+This change essentially combines [persons](/manual/persons) and [events](/manual/events) into a single ClickHouse table, adding person IDs and properties _onto_ events. This change also applies to groups, adding group properties and aggregation to the same table. 
+
+You won’t see any UI changes — persons will still have their own Persons & Groups section on the sidebar, for example — and there’ll be no loss of functionality, but here are some changes you may see:
+
+- **Faster results for queries involving person or group properties and events**. Putting persons, groups and events into a single ClickHouse table means we no longer have to join tables to get results on queries involving these data. As a result, query performance will improve by up to 400%.
+
+- **Faster filtering of events with person or group properties**. Similarly, filtering events by person or group properties is much faster when a join is no longer necessary. Anywhere where you’re working with persons or groups, and events, PostHog will be faster. 
+
+- **Users will no longer be merged retroactively in some situations**. Previously, when an identified user logged in from a different browser (thus becoming anonymous) we would end up with separate records for their unidentified and identified behavior. We would then [merge these into a single person](/docs/how-posthog-works/ingestion-pipeline#merging-two-persons), which could create conflicts and [performance issues](/docs/how-posthog-works/ingestion-pipeline#consequences-of-merging). Our new approach avoids this merge and is much more performant. However, this means the anonymous events are attributed to a different, now non-existent person.
```

Comment:
```suggestion
- **Users will no longer be merged retroactively in some situations**. When an identified user logs in from a different browser (thus becoming anonymous) we end up with separate records for their unidentified and identified behavior, which are then merged into a single person. We used to do this using database joins at query time, so that all events were tied to the same person. Now, we simply look at the events when queries are created - meaning that anonymous events are shown as separate, unique persons in insights.
```
@tiina303 Is this correct? As clear as I could make it. 

This is the one point in all this which I'm a little worried about, however. If we're essentially double-counting a bunch of users then I can see that causing our users to have less faith in our data...

Reply:
This is still a bit off specifically which are then merged into a single person. We used to do this using database joins at query time, so that all events were tied to the same person.
We always do the merge - so future events will be tied to the same person (but we don't edit past events and never have). The change is to what we do at query time - previously we joined. Now we just use the values in the events table (which if we merged have a two person_ids before the merge that would show up as 2 unique persons in trends)

also maybe which are then merged into a single person via $identify or $create_alias - as merging only happens if someone uses one of those two.

Reply:
@tiina303 Can you suggest how you'd change this? This is a little tricky for me to understand and, after re-reading the two versions a few times I'm not 100% clear on what the difference you're suggesting we clarify is.